pkg/common/peertracker/peertracker_test.go

```diff
@@ -1,208 +1,206 @@
-//go:build !windows
-// +build !windows
-
 package peertracker
 
 import (
 	"fmt"
 	"net"
 	"os"
 	"os/exec"
-	"path"
+	"path/filepath"
 	"runtime"
 	"strconv"
 	"syscall"
 	"testing"
 
 	"github.com/sirupsen/logrus"
 	logtest "github.com/sirupsen/logrus/hooks/test"
-	"github.com/spiffe/spire/test/spiretest"
-	"github.com/stretchr/testify/suite"
+	"github.com/stretchr/testify/require"
 )
 
-func TestPeerTrackerTestSuite(t *testing.T) {
-	suite.Run(t, new(PeerTrackerTestSuite))
-}
-
-type PeerTrackerTestSuite struct {
-	suite.Suite
-
+type peertrackerTest struct {
 	childPath string
-	ul        *Listener
-	unixAddr  *net.UnixAddr
+	listener  *Listener
+	addr      net.Addr
 	logHook   *logtest.Hook
 }
 
-func (p *PeerTrackerTestSuite) SetupTest() {
-	tempDir := spiretest.TempDir(p.T())
-
-	p.childPath = path.Join(tempDir, "child")
-	buildOutput, err := exec.Command("go", "build", "-o", p.childPath, "peertracker_test_child.go").CombinedOutput() //nolint: gosec // false positive
+func setupTest(t *testing.T) *peertrackerTest {
+	childPath := filepath.Join(t.TempDir(), "child.exe")
+	buildOutput, err := exec.Command("go", "build", "-o", childPath, childSource).CombinedOutput()
 	if err != nil {
-		p.T().Logf("build output:\n%v\n", string(buildOutput))
-		p.FailNow("failed to build test child")
-	}
-
-	p.unixAddr = &net.UnixAddr{
-		Net:  "unix",
-		Name: path.Join(tempDir, "test.sock"),
+		t.Logf("build output:\n%v\n", string(buildOutput))
+		require.FailNow(t, "failed to build test child")
 	}
 
 	log, hook := logtest.NewNullLogger()
-	p.logHook = hook
+	logHook := hook
 
-	p.ul, err = (&ListenerFactory{Log: log}).ListenUnix(p.unixAddr.Network(), p.unixAddr)
-	p.NoError(err)
-}
+	listener := listener(t, log, addr(t))
 
-func (p *PeerTrackerTestSuite) TearDownTest() {
-	// only close the listener if we haven't already
-	if p.ul != nil {
-		err := p.ul.Close()
-		p.NoError(err)
+	return &peertrackerTest{
+		childPath: childPath,
+		listener:  listener,
+		addr:      listener.Addr(),
+		logHook:   logHook,
 	}
+}
 
-	err := os.Remove(p.childPath)
-	p.NoError(err)
+func (p *peertrackerTest) cleanup(t *testing.T) {
+	if p.listener != nil {
+		err := p.listener.Close()
+		require.NoError(t, err)
+	}
 }
 
-func (p *PeerTrackerTestSuite) TestTrackerClose() {
-	p.ul.Tracker.Close()
-	_, err := p.ul.Tracker.NewWatcher(CallerInfo{})
-	p.Error(err)
+func TestTrackerClose(t *testing.T) {
+	test := setupTest(t)
+	defer test.cleanup(t)
```

Comment:
nit: setupTest can leverage [Cleanup](https://pkg.go.dev/testing#T.Cleanup) to automatically clean up and avoid the need to defer cleanup here.